src/build.rs

```diff
@@ -49,11 +50,39 @@ fn rustc_minor_version() -> Option<u32> {
     otry!(pieces.next()).parse().ok()
 }
 
+fn check_wasm32_target() -> Result<process::ExitStatus, ()> {
+    let echo_cmd = Command::new("echo")
+        .arg("fn main(){}")
+        .stdout(Stdio::piped())
+        .spawn();
+
+    echo_cmd.map_err(|_| ()).and_then(|echo_cmd| {
+        echo_cmd.stdout.ok_or_else(|| ()).and_then(|stdout| {
```

Comment:
If there isn't a `stdout` then `std` is pretty broken and it probably doesn't even make sense to keep working with these child processes anymore. So we should just `stdout.unwrap()` here.

Personally, I also think this would read a bit better using normal `?` error propagation and a return type of `Result<bool, failure::Error>` where the `bool` is whether we have a working `wasm32-unknown-unknown` or not:

```rust
let echo_cmd = Command::new("echo")
    .arg("fn main() {}")
    .stdout(Stdio::piped())
    .spawn()
    .context("failed to spawn `echo` when checking wasm32-unknown-unknown target")?;

let status = Command::new("rustc")
    ...
    .stdin(echo_cmd.stdout.unwrap())
    .status();
match status {
    Ok(s) if s.success() => Ok(true),
    _ => Ok(false),
}
```

Reply:
Thank you for this great review! All of your comments makes sense!